java/kotlin-extractor2/src/main/kotlin/entities/Types.kt

```diff
@@ -1,12 +1,13 @@
 package com.github.codeql
 
+import org.jetbrains.kotlin.analysis.api.symbols.KaClassSymbol
 import org.jetbrains.kotlin.analysis.api.types.KaClassType
 import org.jetbrains.kotlin.analysis.api.types.KaType
 
 private fun KotlinUsesExtractor.useClassType(
     c: KaClassType
 ): TypeResults {
-    val javaResult = addClassLabel(c)
+    val javaResult = TypeResult(addClassLabel(c.symbol as KaClassSymbol) /* , TODO, TODO */)
```

Comment:
Can this `as` cast fail? In what cases?

Reply:
Yes; it's actually a `KaClassLikeSymbol`. I've added a TODO comment for it.